pkg/injector/pod_patch.go

```diff
@@ -271,6 +281,59 @@ func profilingEnabled(annotations map[string]string) bool {
 	}
 }
 
+func appendQuantityToResourceList(quantity string, resourceName v1.ResourceName, resourceList v1.ResourceList) (*v1.ResourceList, error) {
+	q, err := resource.ParseQuantity(quantity)
+	if err != nil {
+		return nil, err
+	}
+	resourceList[resourceName] = q
+	return &resourceList, nil
+}
+
+func getResourceRequirements(annotations map[string]string) (*v1.ResourceRequirements, error) {
+	r := v1.ResourceRequirements{
+		Limits:   v1.ResourceList{},
+		Requests: v1.ResourceList{},
+	}
+	cpuLimit, ok := annotations[daprCPULimitKey]
+	if ok {
+		list, err := appendQuantityToResourceList(cpuLimit, v1.ResourceCPU, r.Limits)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing sidecar cpu limit: %s", err)
+		}
+		r.Limits = *list
+	}
+	memLimit, ok := annotations[daprMemoryLimitKey]
+	if ok {
+		list, err := appendQuantityToResourceList(memLimit, v1.ResourceMemory, r.Limits)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing sidecar memory limit: %s", err)
+		}
+		r.Limits = *list
+	}
+	cpuRequest, ok := annotations[daprCPURequestKey]
```

Comment:
Is the assumption here that its users responsibility to not put any unrealistic request(which can't be filled by any node)?

Reply:
Yes, that's generally how it works on k8s.